nixos/modules/virtualisation/docker.nix

```diff
@@ -193,6 +193,12 @@ in
             ""
             "${pkgs.procps}/bin/kill -s HUP $MAINPID"
           ];
+
+          # Fix from https://github.com/containerd/containerd/pull/4475
+          # (see also https://github.com/containerd/containerd/pull/7566 )
+          # Setting this to infinity breaks e.g. cupsd inside docker
+          # So we reset to the default value.
+          LimitNOFILE = 1048576;
```

Comment:
`LimitNOFILE=1048576` isn't the default in systemd.
`LimitNOFILE=1024:524288` is. Which in turn translates to `LimitNOFILE=524288` and `LimitNOFILESoft=1024`.

The fix `containerd` and `moby/moby` eventually settled on isn't to set it back to `1048576` (which it was before someone set it to `infinity`).
Instead, they both unset that line, resulting in systemd's default value (`1024:524288`).
And the fixed `openrc` and `sysvinit` service files in went with `524288` as well.

See https://github.com/moby/moby/pull/45534/files.

We could also just remove the offending line from the unit file in our `docker_24` package instead of the `nixos/docker` module.
Especially given this will be resolved with `docker_25`, because https://github.com/moby/moby/pull/45534 has been merged already and is even part of the current release candidates.

Speaking of `docker_25`, #278607 carries a patch to set `LimitNOFILE=1048576` for `nixos/docker` as well.

I don't see why we would want to restore the previous `LimitNOFILE=1048576` behavior.
The reasoning and history outlines in the opening comment https://github.com/moby/moby/pull/45534#issue-1708467087 sounds... well... reasonable to me.

I would much rather prefer fixing this in our `docker_24` package, instead of the module.
- If we go with `LimitNOFILE=1048576` in the module, we are stuck with non-default legacy behavior.
- If we go with `LimitNOFILE=1024:524288` in the module, we override the out-of-the-box value for `docker_20_10`.
- If we remove the `LimitNOFILE=infinity` from `${docker_24}/etc/systemd/system/docker.service`, we fix and solve the upstream bug and don't influence the value of `docker_20_10` and the upcoming `docker_25`.